modules/autoscaling-group/main.tf

```diff
@@ -19,7 +19,7 @@ module "asg" {
   vpc_zone_identifier             = var.subnets
 
   # Launch template
-  create_launch_template      = true
+  create_launch_template      = var.create_launch_template && var.launch_type == "EC2"
```

Comment:
@shashimal ,if Launch type is EC2, there should definitely be a Launch_template. I feel condition on launch_type alone is enough.